homeassistant/components/qbittorrent/sensor.py

```diff
@@ -5,121 +5,191 @@
-        return "up_down"
-    if upload > 0 and download == 0:
-        return "seeding"
-    if upload == 0 and download > 0:
-        return "downloading"
-    return STATE_IDLE
+@dataclass(frozen=True, kw_only=True)
+class QBittorrentSensorEntityDescription(SensorEntityDescription):
+    """Entity description class for qBittorent sensors."""
 
-
-def format_speed(speed):
-    """Return a bytes/s measurement as a human readable string."""
-    kb_spd = float(speed) / 1024
-    return round(kb_spd, 2 if kb_spd < 0.1 else 1)
+    val_func: Callable[[QBittorrentDataCoordinator], StateType]
+    extra_state_attr_func: Callable[[Any], dict[str, str]] | None = None
 
 
 SENSOR_TYPES: tuple[QBittorrentSensorEntityDescription, ...] = (
     QBittorrentSensorEntityDescription(
         key=SENSOR_TYPE_CURRENT_STATUS,
+        translation_key="current_status", 
         name="Status",
-        value_fn=_get_qbittorrent_state,
+        device_class=SensorDeviceClass.ENUM,
+        options=[STATE_IDLE, STATE_UP_DOWN, STATE_SEEDING, STATE_DOWNLOADING],
+        val_func=lambda coordinator: get_state(
+            coordinator.data["server_state"]["up_info_speed"], coordinator.data["server_state"]["dl_info_speed"]
+        ),
     ),
     QBittorrentSensorEntityDescription(
         key=SENSOR_TYPE_DOWNLOAD_SPEED,
-        name="Down Speed",
+        translation_key="download_speed",
+        name="Download Speed",
         icon="mdi:cloud-download",
         device_class=SensorDeviceClass.DATA_RATE,
-        native_unit_of_measurement=UnitOfDataRate.KIBIBYTES_PER_SECOND,
-        state_class=SensorStateClass.MEASUREMENT,
-        value_fn=lambda data: format_speed(data["server_state"]["dl_info_speed"]),
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        suggested_display_precision=2,
+        suggested_unit_of_measurement=UnitOfDataRate.MEGABYTES_PER_SECOND,
+        val_func=lambda coordinator: float(coordinator.data["server_state"]["dl_info_speed"]),
     ),
     QBittorrentSensorEntityDescription(
         key=SENSOR_TYPE_UPLOAD_SPEED,
-        name="Up Speed",
+        translation_key="upload_speed",
+        name="Upload Speed",
         icon="mdi:cloud-upload",
         device_class=SensorDeviceClass.DATA_RATE,
-        native_unit_of_measurement=UnitOfDataRate.KIBIBYTES_PER_SECOND,
-        state_class=SensorStateClass.MEASUREMENT,
-        value_fn=lambda data: format_speed(data["server_state"]["up_info_speed"]),
+        native_unit_of_measurement=UnitOfDataRate.BYTES_PER_SECOND,
+        suggested_display_precision=2,
+        suggested_unit_of_measurement=UnitOfDataRate.MEGABYTES_PER_SECOND,
+        val_func=lambda coordinator: float(coordinator.data["server_state"]["up_info_speed"]),
+    ),
+    QBittorrentSensorEntityDescription(
+        key=SENSOR_TYPE_ALL_TORRENTS,
+        translation_key="all_torrents",
+        name="All Torrents",
+        native_unit_of_measurement="Torrents",
```

Comment:
```suggestion
        native_unit_of_measurement="torrents",
```

Reply:
Ah didn't know we should lowercase these, good callout